### PR TITLE
Implement ping referral functionality

### PR DESCRIPTION
**Pull request by ice-atlas**

### CODE REVIEW

File: src/components/ListItems/UserListItem/components/UserListPingButton/index.tsx [hunk 1]
```diff
@@ -1,26 +1,46 @@
...
+  const dispatch = useDispatch();
+
+  const isPingInProgress = useSelector(
+    isLoadingSelector.bind(null, ReferralsActions.PING_REFERRAL(userId)),
+  );
+  const onPress = () => {
+    dispatch(
+      ReferralsActions.PING_REFERRAL(userId).START.create({
+        userId,
+        referralType,
```
Comment (ice-ares) [inline src/components/ListItems/UserListItem/components/UserListPingButton/index.tsx:29]: do u need this>?

File: src/screens/Team/components/SearchResults/index.tsx [hunk 2]
```diff
@@ -68,32 +74,42 @@ export const SearchResults = memo(() => {
...
+        } else if ('id' in item) {
+          return (
+            <UserListItem
+              key={item.id}
+              user={item}
+              AdditionalInfoComponent={
+                item.pinged != null && (
+                  <UserListPingButton
+                    userId={item.id}
+                    referralType={
```
Comment (ice-ares) [inline src/screens/Team/components/SearchResults/index.tsx:98]: why do u need this?

File: src/screens/Team/components/TierList/index.tsx [hunk 3]
```diff
@@ -82,16 +82,25 @@ export const TierList = memo(
...
+    const renderItem = useCallback(
+      ({item}: {item: User}) => {
+        return (
+          <UserListItem
+            user={item}
+            AdditionalInfoComponent={
+              item.pinged != null && (
+                <UserListPingButton
+                  userId={item.id}
+                  referralType={referralType}
```
Comment (ice-ares) [inline src/screens/Team/components/TierList/index.tsx:94]: why do u need this?

File: src/store/modules/Referrals/actions/index.ts [hunk 4]
```diff
@@ -40,7 +32,22 @@ const GET_REFERRALS_HISTORY = createAction('GET_REFERRALS_HISTORY', {
   }),
 });
 
+const PING_REFERRAL = createAction(
+  'REFERRALS/PING_REFERRAL',
+  {
+    START: (payload: {referralType: ReferralType; userId: string}) => payload,
```
Comment (ice-ares) [inline src/store/modules/Referrals/actions/index.ts:38]: why do u need ReferralType?

File: src/store/modules/Referrals/reducer/index.ts [hunk 5]
```diff
@@ -31,35 +31,45 @@ function reducer(state = INITIAL_STATE, action: Actions): State {
...
         draft.history = action.payload.history;
         break;
-      }
-      case AccountActions.SIGN_OUT.SUCCESS.type: {
+
+      case ReferralsActions.PING_REFERRAL(null).SUCCESS.type:
+        {
+          const {userId, referralType} = action.payload;
+
+          const user = draft.data[referralType]?.referrals.find(
```
Comment (ice-ares) [inline src/store/modules/Referrals/reducer/index.ts:59]: oh, now I see why you needed that. No. The one in contacts that can be pinged will also appear on the T1 screen, so those rows should share the same state, If u ping the same user on contacts screen, ping should automatically be disabled for that exact user in the T1 section (based on userID ofc)

File: src/store/modules/Referrals/sagas/pingUserSaga.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  const {referralType, userId} = action.payload;
+  try {
+    yield Api.notifications.pingUser({
+      userId,
+    });
+
+    yield put(
+      ReferralsActions.PING_REFERRAL(action.id).SUCCESS.create({
+        userId,
+        referralType,
```
Comment (ice-ares) [inline src/store/modules/Referrals/sagas/pingUserSaga.ts:20]: u don't need this. match the user on userID

File: src/store/modules/Referrals/reducer/index.ts (hunk 5 above)
Comment (ice-atlas) [reply src/store/modules/Referrals/reducer/index.ts:59]: OK. Now I see. Just used previous implementation of state. Need to redesign it

File: src/screens/HomeFlow/Home/components/Team/components/TeamMember.tsx [hunk 7]
```diff
@@ -6,47 +6,49 @@ import {COLORS} from '@constants/colors';
...
+  return (
+    <Touchable onPress={() => navigation.navigate('UserProfile', {userId})}>
+      <View>
+        <Avatar
+          uri={profilePictureUrl}
+          size={rem(60)}
+          borderRadius={rem(20)}
+          allowFullScreen={false}
+        />
+        {Boolean(phoneNumber) && (
```
Comment (ice-hector) [inline src/screens/HomeFlow/Home/components/Team/components/TeamMember.tsx:40]: !!phoneNumber ?

**Review by ice-hector — APPROVED** [review]

**Review by ice-orion — DISMISSED** [review]

**Review by ice-orion — APPROVED** [review]

**Review by ice-hector — DISMISSED** [review]

**Review by ice-orion — APPROVED** [review]

**Review by ice-hector — APPROVED** [review]

**Review by ice-aiolos — APPROVED** [review]

**Pull request merged**